tests/core/test_processor.py

```diff
@@ -1881,3 +1884,29 @@ async def test_processor_fetch_full_tracker_with_initial_session_existing_tracke
     tracker = await default_processor.fetch_full_tracker_with_initial_session(sender_id)
     assert tracker.sender_id == sender_id
     assert all([event in expected_events for event in tracker.events])
+
+
+@pytest.mark.parametrize("pipeline", [None, "mocked_pipeline"])
+async def test_run_anonymization_pipeline(
+    pipeline: Optional[Text],
+    monkeypatch: MonkeyPatch,
+    default_agent: Agent,
+) -> None:
+    processor = default_agent.processor
+    sender_id = uuid.uuid4().hex
+    tracker = await processor.tracker_store.get_or_create_tracker(sender_id)
+
+    manager = plugin_manager()
+    monkeypatch.setattr(
+        manager.hook, "get_anonymization_pipeline", MagicMock(return_value=pipeline)
+    )
+    event_diff = MagicMock()
+    monkeypatch.setattr(
+        "rasa.shared.core.trackers.TrackerEventDiffEngine.event_difference", event_diff
+    )
+    await processor.run_anonymization_pipeline(tracker)
+
+    if pipeline is not None:
```

Comment:
Tests should not use if and else branches. That indicates that one test covers two different test scenarios. One test should cover one test scenario.